lib/rules/jsx-fragments.js

```diff
@@ -0,0 +1,179 @@
+          + closeFragLong.length - sourceCode.getText(jsxFragment.closingFragment).length;
+        const range = jsxFragment.range;
+        return fixer.replaceTextRange(range, source.slice(range[0], range[1] + lengthDiff));
+      };
+    }
+
+    function getFixerToShort(jsxElement) {
+      return function(fixer) {
+        let source = sourceCode.getText();
+        source = replaceNode(source, jsxElement.closingElement, closeFragShort);
+        source = replaceNode(source, jsxElement.openingElement, openFragShort);
+        const lengthDiff = sourceCode.getText(jsxElement.openingElement).length - openFragShort.length
+          + sourceCode.getText(jsxElement.closingElement).length - closeFragShort.length;
+        const range = jsxElement.range;
+        return fixer.replaceTextRange(range, source.slice(range[0], range[1] - lengthDiff));
+      };
+    }
+
+    function refersToReactFragment(name) {
+      const variableInit = variableUtil.findVariableByName(context, name);
+      if (!variableInit) {
+        return false;
+      }
+
+      // const { Fragment } = React;
+      if (variableInit.type === 'Identifier' && variableInit.name === reactPragma) {
+        return true;
+      }
+
+      // const Fragment = React.Fragment;
+      if (
+        variableInit.type === 'MemberExpression'
+        && variableInit.object.type === 'Identifier'
+        && variableInit.object.name === reactPragma
+        && variableInit.property.type === 'Identifier'
+        && variableInit.property.name === fragmentPragma
+      ) {
+        return true;
+      }
+
+      // const { Fragment } = require('react');
+      if (
+        variableInit.callee
+        && variableInit.callee.name === 'require'
+        && variableInit.arguments
+        && variableInit.arguments[0]
+        && variableInit.arguments[0].value === 'react'
+      ) {
+        return true;
+      }
+
+      return false;
+    }
+
+    const jsxElements = [];
+    const fragmentNames = [`${reactPragma}.${fragmentPragma}`];
+
+    // --------------------------------------------------------------------------
+    // Public
+    // --------------------------------------------------------------------------
+
+    return {
+      JSXElement(node) {
+        jsxElements.push(node);
+      },
+
+      JSXFragment(node) {
+        if (reportOnReactVersion(node)) {
+          return;
+        }
+
+        if (configuration === 'element') {
+          context.report({
+            node,
+            message: `Prefer ${reactPragma}.${fragmentPragma} over fragment shorthand`,
+            fix: getFixerToLong(node)
+          });
+        }
+      },
+
+      ImportDeclaration(node) {
+        if (node.source && node.source.value === 'react') {
+          node.specifiers.forEach(spec => {
+            if (spec.imported && spec.imported.name === fragmentPragma) {
+              if (spec.local) {
+                fragmentNames.push(spec.local.name);
```

Comment:
Is it worth using a Set here instead of an array?

Reply:
Sure - that would communicate the intent better (and probably make no difference performance-wise given the data size here). Just haven't gotten used to the new collections